modules/os-eap-datasource/1.0/added/launch/datasource-common.sh

```diff
@@ -307,6 +310,17 @@ function generate_external_datasource() {
            </validation>"
   fi
 
+  if [ -n "$conn_properties" ]; then
+    properties_array=$(echo $conn_properties | tr ';' '\n' )
```

Comment:
`properties_array=...` This is not creating an array, but rather string.

To get an array,  either `declare -a properties_array=....`, or enclose the subcommand on the right side of the assignment with another pair of the brackes -- IOW: `properties_array=($(my_subcommand_here arg1 ... argN))`

If the variable doesn't need to be an array, you can keep it as is, but drop the `_array` suffix from it, so it's not confusing.

From the above choose the alternative, which works best for you.

Reply:
you are right :+1